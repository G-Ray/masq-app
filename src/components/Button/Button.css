.Button {
  width: 225px;
  height: 50px;
  opacity: 0.8;
  outline: none;
  border-radius: 25px;
  font-weight: 600;
  border: solid 2px #458bf8;
  cursor: pointer;
}

.Button.primary {
  color: white;
  background-color: #458bf8;
  transition: box-shadow 0.2s ease-in-out;
}

.Button.primary:hover {
  box-shadow: 0 10px 16px 0 rgba(0, 0, 0, 0.16), 0 10px 16px 0 rgba(69, 139, 248, 0.4);
  transition: box-shadow 0.2s ease-in-out;
}

.Button.secondary {
  color: #458bf8;
  background-color: white;
}
