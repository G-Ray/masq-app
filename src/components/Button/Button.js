import React from 'react'

import './Button.css'

export default function Button (props) {
  const classes = 'Button ' + (props.secondary ? 'secondary' : 'primary')
  return (
    <button className={classes}>
      {props.label}
    </button>
  )
}
