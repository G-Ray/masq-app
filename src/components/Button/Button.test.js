import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Button from './Button'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Button label='somebutton' />, div)
})

it('renders primary button', () => {
  const component = renderer.create(<Button label='somebutton' />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders secondary button', () => {
  const component = renderer.create(<Button label='somebutton' secondary />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
