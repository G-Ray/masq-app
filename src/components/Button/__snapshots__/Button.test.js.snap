// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders primary button 1`] = `
<button
  className="Button primary"
  onClick={undefined}
>
  somebutton
</button>
`;

exports[`renders secondary button 1`] = `
<button
  className="Button secondary"
  onClick={undefined}
>
  somebutton
</button>
`;
